### ISSUE TITLE
Disable upgrade warnings by default

### ISSUE THREAD

**Issue by iMacTia** [lostisland/faraday#1410]
This seems to be an issue again on the jump from v1 to v2. I'm getting a bunch of these:

```
WARNING: `Faraday::Connection#basic_auth` is deprecated; it will be removed in version 2.0.
While initializing your connection, use `#request(:basic_auth, ...)` instead.
See https://lostisland.github.io/faraday/middleware/authentication for more usage info.
WARNING: `Faraday::Connection#basic_auth` is deprecated; it will be removed in version 2.0.
[...]
```

Did you move away from the `FARADAY_DEPRECATE` idea? To be clear, I don't even directly depend on faraday-- it comes through another dependency.

_Originally posted by @kyrofa in https://github.com/lostisland/faraday/issues/1089#issuecomment-1086227010_

**Comment by hyuraku** [comment]
@iMacTia 
We would like to add `Faraday#DeprecatedClass` to `Faraday ver1.xx` like https://github.com/lostisland/faraday/commit/6e746fceefe0d1f75535b40a62d846a6a95bf85f , How about?

**Comment by iMacTia** [comment]
@hyuraku yep that would work!

**Comment by iMacTia** [comment]
Oh right, `Faraday#DeprecatedClass` deprecates an entire class, but I thought you wanted to use [`Faraday::Deprecate#deprecate`](https://github.com/lostisland/faraday/commit/6e746fceefe0d1f75535b40a62d846a6a95bf85f#diff-a256f6821a10eccdab286b51f2b9e6b178492c404e2a33b08940fe7a0fd75205R70) from that PR. That should allow you do deprecate individual methods with a similar configurable behaviour.

I'd rather not extend or mess with public standard classes like Warning as that could have unexpected repercussions on projects and affects things outside of Faraday's control.

**Comment by iMacTia** [comment]
This was fixed in #1438 🎉 
Thank you @hyuraku 🙏
